internal/provider/instance_resource.go

```diff
@@ -443,6 +453,11 @@ func (r *InstanceResource) Update(ctx context.Context, req resource.UpdateReques
 		body.Volumes = &volumeIds
 	}
 
+	if !data.DiskSize.IsNull() {
+		body.DiskSize = new(int)
+		*body.DiskSize = int(data.DiskSize.ValueInt64())
```

Comment:
```suggestion
		diskSize := pointer(int(data.DiskSize.ValueInt64()))
		body.DiskSize = diskSize
```

Reply:
Maybe you can do it this way for consistency in the codebase. But you code also works.